.github/workflows/run-tests.yml

```diff
@@ -44,7 +44,7 @@ jobs:
         uses: ./.github/actions/setup-dependencies
 
       - name: Check - pyright
-        run: PYRIGHT_PYTHON_FORCE_VERSION=1.1.188 poetry run pyright .
+        run: PYRIGHT_PYTHON_FORCE_VERSION=1.1.307 poetry run pyright .
```

Comment:
If we set a new version which tracks the upstream pyright version in dev dependencies, we no longer need this variables.